### PR TITLE
Add invite links when no contacts

### PR DESCRIPTION
**Pull request by marienfressinaud**
Fix https://github.com/diaspora/diaspora/issues/6062
Continue work from https://github.com/diaspora/diaspora/pull/6189

### CODE REVIEW

**Comment by marienfressinaud** [comment]
Finally I removed link for searching since it was not relevant to be redirected to an empty search page.

**Comment by jhass** [comment]
Could you squash your commits into one and reword Fla's?

```sh
git remote add upstream https://github.com/diaspora/diaspora.git
git fetch upstream
git checkout 6062-add-search-and-invte-links-when-no-contacts
git rebase -i upstream/develop
# Leave the first commit at pick, choose squash or fix for the others.
# If there are any conflicts git will inform you about that, resolve them for
# example with git mergetool or manually and with git add, then run
# git rebase --continue, again  watch the output since that might happen for
# each commit.
git log # Look at the updated commit history
git status # should report you're back on your branch
# We need to force push since a rebase edits the commits, effectively
# creating new ones. This is okay to do in a feature branch, the pull
# request will update.
git push -f origin 6062-add-search-and-invte-links-when-no-contacts # Assuming you cloned from your fork
```

Hint: You can also edit your last commit with `git commit --amend`, since that also effectively creates a new commit, you to need to force push afterwards and again you should only do it in feature branches.

**Comment by marienfressinaud** [comment]
Done

**Comment by jhass** [comment]
https://travis-ci.org/diaspora/diaspora/jobs/84776533#L444

**Comment by marienfressinaud** [comment]
I plan to change test selectors by `.btn.invitations-link` which is working but I'm not sure it's good practice: would you prefer I add two different `id` (one on the stream link, one on the sidebar link) and create cucumber tests for both of them?

File: app/views/shared/_invitations.haml [hunk 1]
```diff
@@ -1,15 +1,9 @@
-= t('.share_this')
+= t(".share_this")
 = invite_link(current_user.invitation_code)
-#invitations-button.btn.btn-link{ "data-toggle" => "modal"}
+.invitations-link.btn.btn-link{"data-toggle" => "modal"}
```
Comment (svbergerem) [inline app/views/shared/_invitations.haml]: You could just add the id `invitations-button` here while keeping the `.invitations-link` class.

**Comment by jhass** [comment]
Updating the selectors for the existing test should be fine I think.

File: app/views/shared/_invitations.haml [hunk 2]
```diff
@@ -1,15 +1,9 @@
-= t('.share_this')
+= t(".share_this")
 = invite_link(current_user.invitation_code)
-#invitations-button.btn.btn-link{ "data-toggle" => "modal"}
+#invitations-button.invitations-link.btn.btn-link{"data-toggle" => "modal"}
```
Comment (diaspora-code-review) [inline app/views/shared/_invitations.haml]: Classes should be listed before IDs (.invitations-link should precede #invitations-button)

Comment (marienfressinaud) [inline app/views/shared/_invitations.haml]: Rhaaa, sorry ><

Comment (jhass) [inline app/views/shared/_invitations.haml]: You can check those locally with `bin/pronto -c develop` btw ;)

**Comment by jhass** [comment]
Mmh, I'd still kept that in the same commit :)

File: app/views/shared/_invitations.haml (hunk 2 above)
Comment (marienfressinaud) [inline app/views/shared/_invitations.haml]: I know but I don't think to it each time. `git commit ; git push` is quicker :)

**Comment by denschub** [comment]
Merged into `develop`. Merging into stable would, unfortunately, be a bit too much effort and it looks like it may break things.

**Pull request merged**